sherpa/stats/tests/test_stats_unit.py

```diff
@@ -497,17 +498,14 @@ def test_stats_calc_stat_wstat_diffbins():
     data, model = setup_single_pha(True, False, background=True)
 
     # Tweak data to have one-less bin than the background
+    counts = data.counts[:-1]
+    staterr = data.staterror[:-1]
+    grouping = data.grouping[:-1]
+
     data.channel = data.channel[:-1]
-    data.counts = data.channel[:-1]
-    for attr in ['staterror', 'syserror', 'grouping', 'quality',
-                 'backscal']:
-        val = getattr(data, attr)
-        if val is not None:
-            try:
-                setattr(data, attr, val[:-1])
-            except TypeError:
-                # assume a scalar, so leave be
-                pass
```

Comment:
Yes, much easier to read now.

Reply:
I am struggling to come up with a reason for why I wrote the code like this in the first place...